AlphaWallet/Swap/SelectServiceToSwapCoordinator.swift

```diff
@@ -0,0 +1,111 @@
+//
+//  SelectServiceToSwapCoordinator.swift
+//  AlphaWallet
+//
+//  Created by Vladyslav Shepitko on 24.08.2022.
+//
+
+import UIKit
+
+protocol SelectServiceToSwapCoordinatorDelegate: AnyObject {
+    func selectSwapService(_ result: Result<SwapTokenUsing, SelectSwapServiceError>, in coordinator: SelectServiceToSwapCoordinator)
+    func didClose(in coordinator: SelectServiceToSwapCoordinator)
+}
+
+class SelectServiceToSwapCoordinator: Coordinator {
+    private let token: Token
+    private let viewController: UIViewController
+    private let swapTokenProvider: SwapTokenProvider
+
+    var coordinators: [Coordinator] = []
+    weak var delegate: SelectServiceToSwapCoordinatorDelegate?
+
+    init(swapTokenProvider: SwapTokenProvider, token: Token, viewController: UIViewController) {
+        self.swapTokenProvider = swapTokenProvider
+        self.token = token
+        self.viewController = viewController
+    }
+
+    func start(wallet: Wallet) {
+        selectBuyService(wallet: wallet, completion: { [token] result in
+            switch result {
+            case .service(let service):
+                if let service = service as? SwapTokenViaUrlProvider {
+                    if let url = service.url(token: token) {
+                        let server = service.rpcServer(forToken: token)
+                        self.delegate?.selectSwapService(.success(.url(url: url, server: server)), in: self)
+                    } else {
+                        self.delegate?.selectSwapService(.failure(SelectSwapServiceError.swapNotSuppoted), in: self)
+                    }
+                } else if let _ = service as? SwapTokenNativeProvider {
+                    let swapPair = SwapPair(from: token, to: nil)
+                    self.delegate?.selectSwapService(.success(.native(swapPair: swapPair)), in: self)
+                } else {
+                    self.delegate?.selectSwapService(.failure(SelectSwapServiceError.swapNotSuppoted), in: self)
+                }
+            case .failure(let error):
+                self.delegate?.selectSwapService(.failure(error), in: self)
+            case .canceled:
+                self.delegate?.didClose(in: self)
+            }
+        })
+    }
+
+    private func selectBuyService(wallet: Wallet, completion: @escaping (SwapTokenUsingService) -> Void) {
```

Comment:
"Buy" should be "Swap" or since it's within the coordinator, maybe just `selectService()`?